Dockerfile

```diff
@@ -15,44 +15,40 @@
 # Build the manager binary
 FROM golang:1.20 as builder
 
-ARG ARCH ?= $(shell go env GOARCH)
-ARG TARGETPLATFORM
-ENV TARGETPLATFORM=${TARGETPLATFORM:-linux/amd64}
-ARG LDFLAGS
-ARG GO_BUILD_GC_ARGS
-
 WORKDIR /workspace
 # Copy the Go Modules manifests
 COPY go.mod go.mod
 COPY go.sum go.sum
 # cache deps before building and copying source so that we don't need to re-download as much
 # and so that source changes don't invalidate our downloaded layer
-RUN go mod download
+RUN --mount=type=cache,target=/go/pkg/mod \
+    go mod download -x
 
 # Copy the go source
-COPY main.go main.go
-COPY api/ api/
-COPY controllers/ controllers/
-COPY pkg/ pkg/
-COPY version/ version/
+COPY ./ ./
 
-# Add kubectl tool
-RUN curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/${TARGETPLATFORM}/kubectl"
-RUN chmod +x ./kubectl
+# Cache the go build into the Go’s compiler cache folder so we take benefits of compiler caching across docker build calls
+RUN --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg/mod \
+    go build main.go
 
-# Add helm chart DIR to get CRDs from it
-COPY deployment/network-operator chart
 # copy CRDs from helm charts
 RUN mkdir crds && \
-    cp -r chart/crds /workspace/crds/network-operator/ && \
-    cp -r chart/charts/sriov-network-operator/crds /workspace/crds/sriov-network-operator/ && \
-    cp -r chart/charts/node-feature-discovery/crds /workspace/crds/node-feature-discovery/
+    cp -r deployment/network-operator/crds /workspace/crds/network-operator/ && \
+    cp -r deployment/network-operator/charts/sriov-network-operator/crds /workspace/crds/sriov-network-operator/ && \
+    cp -r deployment/network-operator/charts/node-feature-discovery/crds /workspace/crds/node-feature-discovery/
+
+ARG LDFLAGS
+ARG GO_BUILD_GC_ARGS
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags="${LDFLAGS}" -gcflags="${GO_BUILD_GC_ARGS}" -a -o manager main.go
```

Comment:
Removing `-a` here saves a lot of time as we can take advantage of go build caching. I'm not sure why we included that option originally though so it might not be appropriate to remove it.